serial/tools/list_ports_linux.py

```diff
@@ -90,11 +90,20 @@ def read_line(self, *args):
 
 def comports(include_links=False, hide_subsystems=["platform"]):
     devices = set()
-    drivers = open('/proc/tty/drivers').readlines()
-    for driver in drivers:
-        items = driver.strip().split()
-        if items[4] == 'serial':
-            devices.update(glob.glob(items[1]+'*'))
+    for driver in open('/proc/tty/drivers'):
+        items = driver.readlines().strip().split()
+        try:
+            if items[4] == 'serial':
+                devices.update(glob.glob(items[1]+'*'))
+        except IndexError:
+            devices.update(glob.glob('/dev/ttyS*'))     # built-in serial ports
```

Comment:
So, as soon as one line in /proc/tty/drivers misbehaves we fall back and add all those randomly chosen devices?
I recommend against this.

Reply:
My goal here is to fall back to the old method in case the drivers are ill-formatted... Although now that you mention it, I _should_ also fall back in case that file is missing for some reason. Beyond that how would you suggest changing the approach? I assume that if one line is malformed, the file isn't trustworthy (i.e. someone went in and manually changed it or something).

Reply:
It is not a real file, but something the kernel just presents as a file.
I assume the contents change the moment you e.g. insert a USB-UART that uses a loadable module driver.
And if someone modifies the Linux kernel on purpose...